src/pynxtools/dataconverter/readers/multi/reader.py

```diff
@@ -0,0 +1,333 @@
+
+    Args:
+        attrs_callback (Callable[[str], Any]):
+            The callback to retrieve attributes under the specified key.
+        data_callback (Callable[[str], Any]):
+            The callback to retrieve the data under the specified key.
+        link_callback (Callable[[str], Any]):
+            The callback to retrieve links under the specified key.
+        eln_callback (Callable[[str], Any]):
+            The callback to retrieve eln values under the specified key.
+        dims (List[str]):
+            The dimension labels of the data. Defaults to None.
+        entry_name (str):
+            The current entry name to use.
+    """
+
+    special_key_map: Dict[str, Callable[[str], Any]]
+    entry_name: str
+    dims: Callable[[], List[str]]
+
+    def __init__(
+        self,
+        attrs_callback: Optional[Callable[[str], Any]] = None,
+        data_callback: Optional[Callable[[str], Any]] = None,
+        link_callback: Optional[Callable[[str], Any]] = None,
+        eln_callback: Optional[Callable[[str], Any]] = None,
+        dims: Optional[Callable[[], List[str]]] = None,
+        entry_name: str = "entry",
+    ):
+        self.special_key_map = {
+            "attrs": attrs_callback if attrs_callback is not None else self.identity,
+            "link": link_callback if link_callback is not None else self.link_callback,
+            "data": data_callback if data_callback is not None else self.identity,
+            "eln": eln_callback if eln_callback is not None else self.identity,
+        }
+
+        self.dims = dims if dims is not None else lambda: []
+        self.entry_name = entry_name
+
+    def link_callback(self, value: str) -> Dict[str, Any]:
+        return {"link": value.replace("/entry/", f"/{self.entry_name}/")}
+
+    def identity(self, value: str) -> str:
+        return value
+
+    def apply_special_key(self, precursor, key, value):
+        """
+        Apply the special key to the value.
+        """
+        return self.special_key_map.get(precursor, self.identity)(value)
+
+
+def parse_json_config(
+    file_path: str,
+    entry_names: List[str],
+    callbacks: Optional[ParseJsonCallbacks] = None,
+) -> dict:
+    """
+    Parses a json file and returns the data as a dictionary.
+    """
+
+    def try_convert(value: str) -> Union[str, float, int, bool]:
+        """
+        Try to convert the value to float, int or bool.
+        If not convertible returns the str itself.
+        """
+
+        if is_integer(value):
+            return int(value)
+        if is_number(value):
+            return float(value)
+        if is_boolean(value):
+            return to_bool(value)
+
+        return value
+
+    if callbacks is None:
+        # Use default callbacks if none are explicitly provided
+        callbacks = ParseJsonCallbacks()
+
+    config_file_dict = parse_flatten_json(file_path)
+    fill_wildcard_data_indices(config_file_dict, callbacks.dims)
```

Comment:
It generally goes through all the keys. But I think the part I forgot is to actually pass an entry name to the `callback.dims` to resolve the dimensions of the current entry.

Reply:
I updated the `get_data_dims` process. It now receives a path which contains the value of the json dict, e.g., `@data:*.index`. This includes the special keyword (because we only resolve it after filling the wildcards).

Also the dims get now called inside the loop, such that `self.callbacks.entry_name` is properly set and can be used inside the `get_data_dims` function.